installation/centos+rhel-setup.sh

```diff
@@ -0,0 +1,172 @@
+
+# general
+pkgs+=" zlib-devel"
+pkgs+=" gettext-devel"
+pkgs+=" libtool-ltdl-devel"
+pkgs+=" libtool-ltdl"
+pkgs+=" glib2-devel"
+pkgs+=" bzip2"
+pkgs+=" m4"
+
+# for yum-config-manager
+pkgs+=" yum-utils"
+
+# runtime dependencies
+pkgs+=" libuuid-devel"
+pkgs+=" libmnl"
+pkgs+=" ${mnl_dev_pkg}"
+pkgs+=" libffi-devel"
+pkgs+=" pcre-devel"
+
+# qemu lite dependencies
+pkgs+=" libattr-devel"
+pkgs+=" libcap-devel"
+pkgs+=" libcap-ng-devel"
+pkgs+=" pixman-devel"
+
+pkgs+=" gcc-c++"
+
+source /etc/os-release
+
+major_version=$(echo "${VERSION_ID}"|cut -d\. -f1)
+
+if [ "${os_distribution}" = rhel ]
+then
+    # RHEL doesn't provide "*-devel" packages unless the "Optional RPMS"
+    # repository is enabled. However, to make life fun, there isn't a
+    # clean way to determine if that repository is enabled (the output
+    # format of "yum repolist" seems to change frequently and
+    # subscription-manager's output isn't designed for easy script
+    # consumption).
+    #
+    # Therefore, the safest approach seems to be to check if a known
+    # required development package is known to yum(8). If it isn't, the
+    # user will need to enable the extra repository.
+    #
+    # Note that this issue is unique to RHEL: yum on CentOS provides
+    # access to developemnt packages by default.
+
+    yum info "${mnl_dev_pkg}" >/dev/null 2>&1
+    if [ $? -ne 0 ]
+    then
+        echo >&2 "ERROR: You must enable the 'optional' repository for '*-devel' packages"
+        exit 1
+    fi
+fi
+
+if [ "${os_distribution}" = centos ]
+then
+    distro="CentOS"
+elif [ "${os_distribution}" = rhel ]
+then
+    distro="RHEL"
+else
+    echo >&2 "ERROR: Unrecognised distribution: ${os_distribution}"
+    echo >&2 "ERROR: This script is designed to work on CentOS and RHEL systems only."
+    exit 1
+fi
+
+site="http://download.opensuse.org"
+dir="repositories/home:/clearcontainers:/clear-containers-3/${distro}_${major_version}"
+repo_file="home:clearcontainers:clear-containers-3.repo"
+cc_repo_url="${site}/${dir}/${repo_file}"
+
+sudo yum -y update
+eval sudo yum -y install "${pkgs}"
+
+sudo yum groupinstall -y 'Development Tools'
+
+pushd "${deps_dir}"
+
+# Install pre-requisites for gcc
+gmp_file="gmp-${gmp_version}.tar.bz2"
+curl -L -O "ftp://gcc.gnu.org/pub/gcc/infrastructure/${gmp_file}"
+compile gmp "${gmp_file}" "gmp-${gmp_version}"
+
+mpfr_file="mpfr-${mpfr_version}.tar.bz2"
+curl -L -O "ftp://gcc.gnu.org/pub/gcc/infrastructure/${mpfr_file}"
+compile mpfr "${mpfr_file}" "mpfr-${mpfr_version}"
+
+mpc_file="mpc-${mpc_version}.tar.gz"
+curl -L -O "ftp://gcc.gnu.org/pub/gcc/infrastructure/${mpc_file}"
+compile mpc "${mpc_file}" "mpc-${mpc_version}"
+
+# Install glib
+glib_setup
+
+# Install json-glib
+json_glib_setup
+
+# Install gcc
+gcc_setup
+
+# Install qemu-lite
+qemu_lite_setup
+
+popd
+
+# Install docker
+sudo yum-config-manager --add-repo https://download.docker.com/linux/centos/docker-ce.repo
+sudo yum -y install docker-ce
```

Comment:
Problem here with RHEL trying to install `docker-ce` with `rhel` repo.

```
$ sudo yum-config-manager --add-repo https://download.docker.com/linux/centos/docker-ce.repo
Loaded plugins: product-id
adding repo from: https://download.docker.com/linux/centos/docker-ce.repo
grabbing file https://download.docker.com/linux/centos/docker-ce.repo to /etc/yum.repos.d/docker-ce.repo
repo saved to /etc/yum.repos.d/docker-ce.repo
+ sudo yum -y install docker-ce
Loaded plugins: product-id, search-disabled-repos, subscription-manager
This system is not registered to Red Hat Subscription Management. You can use subscription-manager to register.
docker-ce-stable                                                                                                                                                                                                     | 2.9 kB  00:00:00
docker-ce-stable/x86_64/primary_db                                                                                                                                                                                   | 7.8 kB  00:00:00
Resolving Dependencies
--> Running transaction check
---> Package docker-ce.x86_64 0:17.06.1.ce-1.el7.centos will be installed
--> Processing Dependency: container-selinux >= 2.9 for package: docker-ce-17.06.1.ce-1.el7.centos.x86_64
--> Processing Dependency: libseccomp.so.2()(64bit) for package: docker-ce-17.06.1.ce-1.el7.centos.x86_64
--> Running transaction check
---> Package docker-ce.x86_64 0:17.06.1.ce-1.el7.centos will be installed
--> Processing Dependency: container-selinux >= 2.9 for package: docker-ce-17.06.1.ce-1.el7.centos.x86_64
---> Package libseccomp.x86_64 0:2.3.1-2.el7 will be installed
--> Finished Dependency Resolution
Error: Package: docker-ce-17.06.1.ce-1.el7.centos.x86_64 (docker-ce-stable)
           Requires: container-selinux >= 2.9
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
```

Reply:
Looks like this works for rhel:

`sudo yum install yum install -y --setopt=obsoletes=0 docker-ce-17.03.1.ce-1.el7.centos docker-ce-selinux-17.03.1.ce-1.el7.centos`

Reply:
Soution:

add this line if the OS is RHEL before installing docker-ce

`sudo rpm -Uvh http://mirror.centos.org/centos/7/extras/x86_64/Packages/container-selinux-2.9-4.el7.noarch.rpm`

Reply:
Although it might resolve the problem, I'm not sure we should be doing this because:

- That command is downloading a security policy package via HTTP (insecure).
- RHEL admins are unlikely to be happy having to install a package using `rpm` directly.
- The package is coming from a CentOS repo, not a RHEL one.
- The package in question may differs between RHEL and CentOS and given its security nature, there is a risk associated in this.

Having found the link below, I think we should (yet again ;-) consider switching the instructions and the script back to being CentOS only:

- https://docs.docker.com/engine/installation/linux/docker-ee/rhel/

@grahamwhaley, @sameo - thoughts?

Reply:
Well, there is no `docker-ce` for RHEL, only `docker-ee` and we do not test CC with that version. 

The script was pointing to install `docker-ce` from CentOS repo so...

Reply:
@jodh-intel agree, let's make this PR CentOS only